### PR TITLE
adds ammo piles

### PR DESCRIPTION
**Pull request by Andromeda-K22**
🆑 Adds ammo piles.
Click on another bit of ammo with a shell in hand, and you'll make an ammo pile. Only works for same-types of ammo right now, because I couldn't figure out a better way to make them spawn properly when created.

🆑

### CODE REVIEW

**Comment by Sparrowbird611** [comment]
Is this the answer to our prayers? No longer shall we have to toil, picking up shell after shell...

This is basically what eris does with ammo, right?

**Comment by Andromeda-K22** [comment]
I think? Maybe?

File: code/modules/boh/guns/ammo_pile.dm [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		var/obj/C = new ammo_type(get_turf(src))
+		C.forceMove(src)
+		ammo += C
+	update_appearance()
+
+/obj/item/weapon/ammo_pile/examine(mob/user)
+	. = ..()
+	to_chat(user, SPAN_NOTICE("It has the following shells in it:"))
+	for(var/obj/item/ammo_casing/C in ammo)
+		to_chat(user, SPAN_NOTICE("\icon[src] [C.name]"))
```
Comment (BurgerLUA) [inline code/modules/boh/guns/ammo_pile.dm]: Wouldn't this spam chat for every single bullet inside it?

File: code/modules/boh/guns/ammo_pile.dm [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+/obj/item/weapon/ammo_pile/proc/check_ammo()
+	if(!LAZYLEN(ammo))
+		qdel(src)
+		return
+	update_appearance()
+
+/obj/item/weapon/ammo_pile/proc/update_appearance()
+	for(var/image/I in overlays)
+		qdel(I)
```
Comment (BurgerLUA) [inline code/modules/boh/guns/ammo_pile.dm]: I'm 90% sure you don't need to qdel images.

File: code/modules/projectiles/ammunition.dm [hunk 3]
```diff
@@ -74,6 +74,16 @@
 		else
 			to_chat(user, "<span class='notice'>You inscribe \"[label_text]\" into \the [initial(BB.name)].</span>")
 			BB.SetName("[initial(BB.name)] (\"[label_text]\")")
+	if(istype(W, src.type))
+		var/obj/item/weapon/ammo_pile/AP = new (get_turf(user))
+		var/obj/item/ammo_casing/C = W
+		AP.name = "[src.name] ammo pile"
+		AP.ammo_type = type
+		AP.CreatePile(2)
```
[the remark this answers is not in the document]
Comment (MarinaGryphon) [reply code/modules/projectiles/ammunition.dm]: Noooo. You have two bullets. They may have special properties like inscriptions, etc. and are already the same type. ADD. THEM. TO. THE. PILE. Don't delete them and make two NEW ones.

[the remark this answers is not in the document]
Comment (Andromeda-K22) [reply code/modules/projectiles/ammunition.dm]: Yeah that doesn't work. Tried it.

**Comment by BurgerLUA** [comment]
Conflicts and some unaddressed code changes. Please fix them.

**Comment by BurgerLUA** [comment]
Is this good to merge/review?

**Comment by BurgerLUA** [comment]
is this good to merge yes or no

**Comment by Andromeda-K22** [comment]
no

**Comment by BurgerLUA** [comment]
Closing this PR as it is on the second page with no recent changed. Re-open it if you wish to work on it.

**Pull request closed without merging**